In [ ]:
!pip install "ipywidgets>=7,<8" -q # https://github.com/googlecolab/colabtools/issues/3020

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 6.5 MB/s eta 0:00:00


In [ ]:
from ipywidgets import HTML, Password, Text, Textarea, VBox
inputs = [
      Password(description="Github Token:"),
      Text(description='Github Email:'),
      Text(description='Github User:'),
]
VBox(inputs)

In [ ]:
token = inputs[0].value
email = inputs[1].value
username = inputs[2].value

github_url_input = Textarea(value=f'https://github.com/{username}/openmapflow.git')
VBox([HTML(value="<b>Github Clone URL</b>"), github_url_input])

In [ ]:
from pathlib import Path

github_url = github_url_input.value
project_name = "crop-mask-example" # maize-example
country_name = "Togo" # Kenya

for input_value in [token, email, username, github_url]:
  if input_value.strip() == "":
    raise ValueError("Found input with blank value.")

path_to_project = f"{Path(github_url).stem}/{project_name}"

!git config --global user.email $username
!git config --global user.name $email
!git clone {github_url.replace("https://", f"https://{username}:{token}@")}

%cd {path_to_project}

!pip install openmapflow[all]==0.2.2rc1 -q
!pip install dvc[gs] cmocean pyyaml==5.4.1 -q

In [ ]:
!pip install dvc[gs] -q

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 67.8/67.8 kB 6.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.8/41.8 kB 3.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.7/46.7 kB 3.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 154.5/154.5 kB 15.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112.9/112.9 kB 13.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.5/57.5 kB 7.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.5/45.5 kB 5.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 512.2/512.2 kB 24.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 421.4/421.4 kB 44.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 196.7/196.7 kB 23.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.5/79.5 kB 9.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 117.0/117.0 kB 15.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ..

In [ ]:
# Installing gdal
%%shell
yes | add-apt-repository ppa:ubuntugis/ppa && apt-get update -qq
apt-get install python3-gdal gdal-bin -y -q

PPA publishes dbgsym, you may need to include 'main/debug' component
Repository: 'deb https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu/ jammy main'
Description:
Official stable UbuntuGIS packages.


More info: https://launchpad.net/~ubuntugis/+archive/ubuntu/ppa
Adding repository.
Found existing deb entry in /etc/apt/sources.list.d/ubuntugis-ubuntu-ppa-jammy.list
Adding deb entry to /etc/apt/sources.list.d/ubuntugis-ubuntu-ppa-jammy.list
Found existing deb-src entry in /etc/apt/sources.list.d/ubuntugis-ubuntu-ppa-jammy.list
Adding disabled deb-src entry to /etc/apt/sources.list.d/ubuntugis-ubuntu-ppa-jammy.list
Adding key to /etc/apt/trusted.gpg.d/ubuntugis-ubuntu-ppa.gpg with fingerprint 6B827C12C2D425E227EDCA75089EBE08314DF160
Get:1 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]
Hit:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease
Hit:4

In [ ]:
# A Google Cloud Account is required to access the data
!gcloud auth application-default login

Go to the following link in your browser:

    https://accounts.google.com/o/oauth2/auth?response_type=code&client_id=764086051850-6qr4p6gpi6hn506pt8ejuq83di341hur.apps.googleusercontent.com&redirect_uri=https%3A%2F%2Fsdk.cloud.google.com%2Fapplicationdefaultauthcode.html&scope=openid+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcloud-platform+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fsqlservice.login&state=QSq07HrXHybXTPLaiDfGJ1K3RqQBn3&prompt=consent&access_type=offline&code_challenge=OyF9XEEf0Gw91uM4lNDjF-Pa5YlqjgOLcUfXxjlMPiU&code_challenge_method=S256

Enter authorization code: 4/0AfJohXnXE1OPu9SeYHL6zY04LVW1R_7OzA9vMjNeodQfBtM3A5TTHAhLSpnw423k1re3yQ

Credentials saved to file: [/content/.config/application_default_credentials.json]

These credentials will be used by any library that requests Application Default Credentials (ADC).
Cannot find a quota project to add to ADC. You might receive a "quota exceeded" or "API not enable

In [ ]:
# Pull in data already available
!dvc pull

In [ ]:
import pandas as pd
import geopandas as gpd
from shapely.geometry import Point
from datasets import datasets, label_col
from openmapflow.constants import LAT, LON, DATASET, SUBSET

In [ ]:
import pandas as pd
df = pd.concat([d.load_df() for d in datasets])

In [ ]:
df.subset.value_counts()

training      35261
testing         309
validation      277
Name: subset, dtype: int64

Togo

Training: 991

Testing: 309

Validation: 277

Total: 1577

México

Training: 539

Validation: 135

Total: 674


In [ ]:
df.to_csv('df_1.csv')

In [ ]:
df_mexico = pd.read_csv('df_mexico.csv')
df_mexico.head(5)

lon        lat  start_date    end_date         source  \
0 -106.75  30.452381  2017-01-01  2018-12-31  loc_all_2.txt   
1 -102.75  18.452381  2017-01-01  2018-12-31  loc_all_2.txt   
2 -111.75  31.452381  2017-01-01  2018-12-31  loc_all_2.txt   
3 -104.75  19.452381  2017-01-01  2018-12-31  loc_all_2.txt   
4 -100.75  18.452381  2017-01-01  2018-12-31  loc_all_2.txt   

   class_probability  num_labelers    subset analysis_duration email country  \
0                0.0             4  training               ,,,   ,,,  global   
1                0.0             5  training              ,,,,  ,,,,  global   
2                0.0             5  training              ,,,,  ,,,,  global   
3                0.0             5  training              ,,,,  ,,,,  global   
4                0.0             4  training               ,,,   ,,,  global   

                  dataset                                            eo_data  \
0  geowiki_landcover_2017  [[-1.43404609e+01 -2.57548950e+01  1.65200000e...   
1  geowiki_landcover_2017  [[ 1.31849083e+00 -7.14152434e+00  8.88000000e...   
2  geowiki_landcover_2017  [[-1.48538073e+01 -2.12222881e+01  1.11700000e...   
3  geowiki_landcover_2017  [[-1.34904642e+01 -1.75946169e+01  1.00300000e...   
4  geowiki_landcover_2017  [[-1.28127582e+01 -1.57214681e+01  8.88000000e...   

       eo_lon     eo_lat                                            eo_file  \
0 -106.749994  30.452394  min_lat=30.4517_min_lon=-106.7508_max_lat=30.4...   
1 -102.749976  18.452339  min_lat=18.4517_min_lon=-102.7508_max_lat=18.4...   
2 -111.750017  31.452399  min_lat=31.4517_min_lon=-111.7508_max_lat=31.4...   
3 -104.749985  19.452344  min_lat=19.4517_min_lon=-104.7508_max_lat=19.4...   
4 -100.749967  18.452339  min_lat=18.4517_min_lon=-100.7508_max_lat=18.4...   

          eo_status  
0  eo_data_complete  
1  eo_data_complete  
2  eo_data_complete  
3  eo_data_complete  
4  eo_data_complete

In [ ]:
df_mexico_upd = df_mexico.sample(135)
df_mexico_upd.subset = 'validation'
df_mexico.update(df_mexico_upd)

/usr/local/lib/python3.10/dist-packages/pandas/core/frame.py:7511: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[col] = expressions.where(mask, this, that)


In [ ]:
df_mexico.subset.value_counts()

training      539
validation    135
Name: subset, dtype: int64

In [ ]:
df_mexico.to_csv('df_mexico_train_val.csv')

In [ ]:
import os
os.environ["MODEL_NAME"] = input("MODEL_NAME=")

MODEL_NAME=test_1


In [ ]:
!python train.py --model_name $MODEL_NAME --epoch 3

Upsampling: positive class from 7807 to 13727 using upsampling ratio: 0.5
Epoch:   0%|          | 0/3 [00:00<?, ?it/s]
MODEL_NAME=test_1
accuracy: 0.7076
f1: 0.6989
precision: 0.8393
recall: 0.5987
roc_auc: 0.8392



In [ ]:
os.environ["MODEL_NAME"] = input("MODEL_NAME=")

MODEL_NAME=test_mex_2


In [ ]:
!python train_mexico.py --model_name $MODEL_NAME --epoch 3 --batch_size 32

Upsampling: positive class from 80 to 230 using upsampling ratio: 0.5
Epoch:   0%|          | 0/3 [00:00<?, ?it/s]
MODEL_NAME=test_mex_2
accuracy: 0.8741
f1: 0.4516
precision: 0.4667
recall: 0.4375
roc_auc: 0.7794



In [ ]:
gdf_puntos_mexico = pd.read_csv('gdf_puntos_mexico.csv')

In [ ]:
mexico_ids = gdf_puntos_mexico['Unnamed: 0'].tolist()

In [ ]:
len(mexico_ids)

674

In [ ]:
df_mexico = df[df.index.isin(mexico_ids)]

In [ ]:
df_mexico.to_csv('df_mexico.csv')